internal/controller/controlplane_controller.go

```diff
@@ -223,17 +179,41 @@ func (r *ControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	// NOW determine overall CP readiness based on BOTH API server AND hooks
-	log.Info("Determining final control plane readiness")
-
-	// Check API server readiness
+	// PHASE 2: Check API server readiness
 	apiServerReady, err := util.IsAPIServerDeploymentReady(log, r.Client, *hcp)
 	if err != nil {
 		log.Error(err, "Error checking API server readiness", "controlPlane", hcp.Name)
 	}
 	log.Info("API server readiness check", "controlPlane", hcp.Name, "apiServerReady", apiServerReady)
 
-	// Determine final readiness based on waitForPostCreateHooks setting
+	// Requeue if API Server is Not Ready
+	if !apiServerReady {
+		log.Info("API Server Not Ready. Requeuing...", "controlPlane", hcp.Name)
+		// Update Status
+		tenancyv1alpha1.EnsureCondition(hcp, tenancyv1alpha1.ConditionUnavailable())
+		if err = r.Status().Update(ctx, hcp); err != nil {
+			log.Error(err, "Failed to update ControlPlane status")
+		}
+		return ctrl.Result{RequeueAfter: time.Second * 15}, nil
+	}
+
+	// PHASE 3: PostCreateHook processing
+	if hcp.Spec.PostCreateHook != nil || len(hcp.Spec.PostCreateHooks) > 0 {
+		log.Info("Processing PostCreateHooks with complete kubeconfig")
+
+		if err := reconciler.ReconcileUpdatePostCreateHook(ctx, hcp); err != nil {
+			log.Error(err, "Failed to process PostCreateHooks")
+			// Don't return error immediately - let status logic handle it
```

Comment:
The error value evaporates at the close bracket on the next line, so no logic will handle it.